notifier/tasks.py

```diff
@@ -59,6 +62,44 @@ def generate_and_send_digests(users, from_dt, to_dt):
             # raise right away, since we don't support partial retry
             raise
 
+@celery.task(rate_limit=settings.FORUM_DIGEST_TASK_RATE_LIMIT, max_retries=settings.FORUM_DIGEST_TASK_MAX_RETRIES)
+def generate_and_send_digests_flagged(messages):
+    """
+    This task generates and sends flagged forum digest emails to multiple users
+    in a single background operation.
+
+    Args:
+        messages (list): contains dicts with the following keys:
+            course_id (str): identifier of the course
+            recipient (dict): a single user dict
+            threads (list): a list of thread URLs
+    """
+    with closing(get_connection()) as cx:
```

Comment:
We should avoid holding this connection open any longer than required.

Since we don't use `cx` until _after_ we've built up our list of messages, could we postpone opening the connection like:


```python
rendered_messages = []
for message in messages:
    # ...
    rendered_messages.append(rendered_message)
with closing(get_connection()) as connection:
    try:
        connection.send_messages(rendered_messages)
    except:
        # ...